engine/evd.ml

```diff
@@ -2055,3 +2075,13 @@ let drop_new_defined ~original sigma =
   let to_keep = normalize_against original.defn_evars to_keep in
   let undf_evars = normalize_against original.undf_evars sigma.undf_evars in
   { sigma with defn_evars = to_keep; undf_evars }
+
+let has_undefined_head_evar sigma c =
```

Comment:
Given that this 1. this shouldn't be declared here because it has access to the full Evd internals and can be defined only with EConstr.kind and 2. is only used once, I think it'd be simpler just to move it internally to Coercion.

Reply:
Done